cmd/kpod/ps.go

```diff
@@ -0,0 +1,367 @@
+		command          string
+		runningFor       string
+		imageName        string
+		mounts           string
+		ports            string
+		size             string
+	)
+
+	if len(containers) > 0 {
+		if opts.latest {
+			containersOutput = append(containersOutput, containers[0])
+		} else if opts.last > 0 {
+			if opts.last > len(containers) {
+				containersOutput = containers
+			} else {
+				for i := 0; i < opts.last; i++ {
+					containersOutput = append(containersOutput, containers[i])
+				}
+			}
+		} else if !opts.all {
+			for _, ctr := range containers {
+				if ctr.State.Status == "running" {
+					containersOutput = append(containersOutput, ctr)
+				}
+			}
+		} else {
+			containersOutput = containers
+		}
+
+		for _, ctr := range containersOutput {
+			ctrID = ctr.ID
+			runningFor = units.HumanDuration(time.Since(ctr.State.Created))
+			command = getCommand(ctr.ImageCreatedBy)
+			imageName = getImageName(ctr.FromImage, opts.noTrunc)
+			mounts = getMounts(ctr.Mounts, opts.noTrunc)
+			ports = getPorts(ctr.Config.ExposedPorts)
+			size = units.HumanSize(float64(ctr.SizeRootFs))
+
+			if ctr.State.Status == "stopped" {
+				status = "Exited (" + string(ctr.State.ExitCode) + ") " + runningFor + " ago"
+			} else if ctr.State.Status == "running" {
+				status = "Up " + runningFor + " ago"
+			} else {
+				status = "Created"
+			}
+
+			if !opts.noTrunc {
+				ctrID = ctr.ID[:idTruncLength]
+			}
+
+			params := psOutputParams{
+				ID:         ctrID,
+				Image:      imageName,
+				Command:    command,
+				CreatedAt:  ctr.State.Created,
+				RunningFor: time.Since(ctr.State.Created),
+				Status:     status,
+				Ports:      ports,
+				Size:       size,
+				Names:      ctr.Name,
+				Labels:     ctr.Labels,
+				Label:      "",
+				Mounts:     mounts,
+			}
+			output = append(output, params)
+		}
+	}
+	var out formats.Writer
+
+	if opts.format != "" {
+		switch opts.format {
+		case "json":
+			out = formats.JSONstruct{Output: toGenericPS(output)}
+		default:
+			out = formats.StdoutTemplate{Output: toGenericPS(output), Template: opts.format}
+		}
+	} else {
+		out = outStruct{output: output, opts: opts}
+	}
+
+	formats.Writer(out).Out()
+	return nil
+}
+
+// getCommand gets the actual command from the whole command
+func getCommand(cmd string) string {
+	start := strings.Index(cmd, "[")
+	if start == -1 {
+		return ""
+	}
+	start += len("[")
+	end := strings.Index(cmd, "]")
+	return cmd[start:end]
+}
+
+// getImageName shortens the image name when noTrunc is false
+func getImageName(img string, noTrunc bool) string {
+	var out string
+	if !noTrunc {
+		arr := strings.Split(img, "/")
+		if arr[0] == "docker.io" {
+			out = strings.Join(arr[2:], "/")
+		} else {
+			out = img
+		}
+	} else {
+		out = img
+	}
+	return out
+}
+
+// getLabels converts the labels to a string of the form "key=value, key2=value2"
+func getLabels(labels fields.Set) string {
+	var arr []string
+	if len(labels) > 0 {
+		for key, val := range labels {
```

Comment:
I don't think so, it is a map and I need both key and value in the format "key=value".